xbmc/filesystem/CurlFile.cpp

```diff
@@ -601,9 +601,15 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
       StringUtils::Format(":%d", m_proxyport);
     g_curlInterface.easy_setopt(h, CURLOPT_PROXY, hostport.c_str());
 
-    const std::string userpass =
-      m_proxyuser + std::string(":") + m_proxypassword;
-    if (userpass != ":")
+    std::string userpass = "";
+
+    if (!m_proxyuser.empty() &&
+        !m_proxypassword.empty())
+      userpass = CURL::Encode(m_proxyuser) +
+          CURL::Encode(":") +
+          CURL::Encode(m_proxypassword);
```

Comment:
All this on one line should be under 100 chars I think.

Reply:
yes